desktop/core/src/desktop/js/reactComponents/TaskBrowser/TaskBrowser.tsx

```diff
@@ -0,0 +1,434 @@
+  const date = new Date(timestamp);
+  return date.toLocaleString('en-US', options);
+}
+
+
+export const TaskBrowserTable = ({ handleShowLogs, handleSchedulePopup }) => {
+
+  const [tasks, setTasks] = useState([]);
+  const [searchTerm, setSearchTerm] = useState(''); // state for the search term
+  const [statusFilter, setStatusFilter] = useState({ success: false, failure: false, running: false, all: true }); // state to track status filter
+  const [selectedTasks, setSelectedTasks] = useState([]); // state for check box of a task
+
+  useEffect(() => {
+    const fetchTasks = () => {
+    axios.get('/desktop/api2/get_taskserver_tasks/')
+        .then(response => {
+            if (Array.isArray(response.data)) {
+            //sort based on timestamp
+            const sortedTasks = response.data.sort((a, b) => {
+                // Assuming 'date_done' is the timestamp field and is in a comparable format
+                return new Date(b.date_done) - new Date(a.date_done);
+            });
+            setTasks(sortedTasks);
+          } else {
+            console.error('Expected an array of tasks, but received:', response.data);
+          }
+        })
+        .catch(error => {
+            console.error('Error fetching tasks', error);
+        });
+    };
+
+    fetchTasks();   //fetch tasks initially
+    const interval = setInterval(fetchTasks, 5000);
+    return () => clearInterval(interval); // Cleanup on unmount
+
+  }, []);
+
+  const handleTaskIdClick = (taskId) => {
+    setCurrentTaskId(taskId);
+    setShowLogsPopup(true);
+    // Here, you could also fetch the logs immediately if you want to
+    fetchTaskLogs(taskId);
+  };
+
+
+
+  const handleSearchChange = (e) => {
+   console.log('Filtering tasks by:', e.target.value);
+    setSearchTerm(e.target.value.toLowerCase());
+    };
+
+  // Function to handle status filter button clicks
+  const handleStatusFilterChange = (status) => {
+    setStatusFilter(prevStatusFilter => {
+      const isAll = status === 'all';
+      const newStatusFilter = {
+        ...prevStatusFilter,
+        [status]: isAll ? true : !prevStatusFilter[status],
+      };
+  
+      if (isAll) {
+        // If 'all' is selected, set everything else to false
+        newStatusFilter.success = false;
+        newStatusFilter.failure = false;
+        newStatusFilter.running = false;
+      } else {
+        // If any specific status is toggled, set 'all' to false
+        newStatusFilter.all = false;
+      }
+  
+      // If no individual statuses are selected, default to 'all'
+      if (!newStatusFilter.success && !newStatusFilter.failure && !newStatusFilter.running) {
+        newStatusFilter.all = true;
+      }
+  
+      return newStatusFilter;
+    });
+  };
+  
+  
+
+  const filteredTasks = tasks.filter(task => {
+    const taskNameMatch = task.result?.task_name?.toLowerCase().includes(searchTerm);
+    const userIdMatch = task.result?.username?.toString().toLowerCase().includes(searchTerm);
```

Comment:
Do you need `toString()` on the username?

Reply:
Not needed. "taskNameMatch" and "userIdMatch" are fetched from "task.result" and they are strings already. Initially I was fetching as an object, so added the toString() method.